[Bug]: Ci Astrobook Build Fail And Broken Prod Astrobook Slack Webhook

### What happened?

Two separate issues occurred:

1. **Build error:** Rollup failed to resolve the import `@oxc-parser/binding-wasm32-wasi`, causing the build process to fail.

2. **Slack webhook issue:** The commit hash and last changes from the PR title are not correctly displayed in the Slack webhook message — they appear incorrect or missing.

### What did you expect to happen?

1. The build should complete successfully without any resolution errors for `@oxc-parser/binding-wasm32-wasi`.

2. The Slack webhook message should display the correct commit hash and the title accurately.

### Steps to reproduce

1. Run the build process for astrobook "just build book"
2. Observe the Rollup error related to `@oxc-parser/binding-wasm32-wasi`
3. Open a PR and check the Slack webhook notification
4. Compare the displayed commit hash and PR title changes with the actual values — they don't match

### Version

1.0.0

### What browsers are you seeing the problem on?

_No response_

### What operating system are you using?

Linux

### Relevant log output

```shell
[ERROR] [vite] ✗ Build failed in 308ms
[vite]: Rollup failed to resolve import "@oxc-parser/binding-wasm32-wasi" from "/home/runner/work/kodaskills.co/kodaskills.co/node_modules/@unocss/transformer-attributify-jsx/node_modules/oxc-parser/src-js/wasm.js".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
  Stack trace:
    at viteLog (file:///home/runner/work/kodaskills.co/kodaskills.co/node_modules/vite/dist/node/chunks/config.js:33635:57)
    at onLog (file:///home/runner/work/kodaskills.co/kodaskills.co/node_modules/vite/dist/node/chunks/config.js:33467:4)
    at Object.logger [as onLog] (file:///home/runner/work/kodaskills.co/kodaskills.co/node_modules/rollup/dist/es/shared/node-entry.js:23023:9)
    at file:///home/runner/work/kodaskills.co/kodaskills.co/node_modules/rollup/dist/es/shared/node-entry.js:21725:26
error: script "build" exited with code 1
Error: Process completed with exit code 1.
```

### Screenshots

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.github/actions/slack-notify/action.yml
+++ b/.github/actions/slack-notify/action.yml
@@ -50,6 +50,14 @@ runs:
           echo "STATUS_EMOJI=🟡" >> $GITHUB_ENV
           echo "COLOR=#ECB22E" >> $GITHUB_ENV
         fi
+        if [ -n "$PR_NUMBER" ]; then
+          echo "LABEL_PREFIX=Last changes from PR" >> $GITHUB_ENV
+          echo "PR_DISPLAY=#$PR_NUMBER — $PR_TITLE" >> $GITHUB_ENV
+        else
+          echo "LABEL_PREFIX=Last changes from main" >> $GITHUB_ENV
+          MAIN_TITLE="$(git show --pretty=format:%s -s HEAD)"
+          echo "PR_DISPLAY=$MAIN_TITLE" >> $GITHUB_ENV
+        fi
         echo "PR_NAME=$PR_TITLE" >> $GITHUB_ENV
         echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
         echo "COMMIT_HASH=$COMMIT_HASH" >> $GITHUB_ENV
@@ -75,7 +83,7 @@ runs:
                     "fields": [
                       {
                         "type": "mrkdwn",
-                        "text": "*Last changes from PR*\n<${{ env.PR_URL }}|#${{ env.PR_NUMBER }} — ${{ env.PR_NAME }}>"
+                        "text": "*${{ env.LABEL_PREFIX }}*\n<${{ env.PR_URL }}|${{ env.PR_DISPLAY }}>"
                       }
                     ]
                   },

--- a/.github/workflows/deploy-astrobook.yml
+++ b/.github/workflows/deploy-astrobook.yml
@@ -54,7 +54,7 @@ jobs:
       url: ${{ steps.deploy.outputs.deployment-url }}
     env:
       DEPLOY_BRANCH: ${{ github.ref_name }}
-      DEPLOY_COMMIT: ${{ github.event.pull_request.head.sha }}
+      DEPLOY_COMMIT: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
       NOTIFY_PR_TITLE: ${{ github.event_name == 'pull_request' && github.event.pull_request.title || github.ref_name }}
       NOTIFY_PR_NUMBER: ${{ github.event_name == 'pull_request' && github.event.pull_request.number || '' }}
       NOTIFY_PR_URL: ${{ github.event_name == 'pull_request' && github.event.pull_request.html_url || format('https://github.com/{0}/commit/{1}', github.repository, github.sha) }}

